Assets/Scripts/Lua/CLRBindings/LuaArenaStatus.cs

```diff
@@ -29,54 +29,50 @@ public class LuaArenaStatus {
+        ArenaManager.instance.Show();
     }
 
     public void Move(float x, float y, bool movePlayer = true, bool immediate = false) {
-        if (UIController.instance.GetState() == UIController.UIState.DEFENDING) {
-            if (immediate) ArenaManager.instance.MoveImmediate(x, y, movePlayer);
-            else           ArenaManager.instance.Move(x, y, movePlayer);
-        } else
-            UnitaleUtil.DisplayLuaError("NOT THE WAVE SCRIPT", "sorry but pls don't");
+        if (immediate) ArenaManager.instance.MoveImmediate(x, y, movePlayer);
+        else           ArenaManager.instance.Move(x, y, movePlayer);
     }
 
     public void MoveTo(float x, float y, bool movePlayer = true, bool immediate = false) {
-        if (UIController.instance.GetState() == UIController.UIState.DEFENDING) {
-            if (immediate) ArenaManager.instance.MoveToImmediate(x, y, movePlayer);
-            else           ArenaManager.instance.MoveTo(x, y, movePlayer);
-        } else
-            UnitaleUtil.DisplayLuaError("NOT THE WAVE SCRIPT", "sorry but pls don't");
+        if (immediate) ArenaManager.instance.MoveToImmediate(x, y, movePlayer);
+        else           ArenaManager.instance.MoveTo(x, y, movePlayer);
     }
 
     public void MoveAndResize(float x, float y, int width, int height, bool movePlayer = true, bool immediate = false) {
-        if (UIController.instance.GetState() == UIController.UIState.DEFENDING) {
-            if (immediate) ArenaManager.instance.MoveAndResizeImmediate(x, y, width, height, movePlayer);
-            else           ArenaManager.instance.MoveAndResize(x, y, width, height, movePlayer);
-        } else
-            UnitaleUtil.DisplayLuaError("NOT THE WAVE SCRIPT", "sorry but pls don't");
+        if (immediate) ArenaManager.instance.MoveAndResizeImmediate(x, y, width, height, movePlayer);
+        else           ArenaManager.instance.MoveAndResize(x, y, width, height, movePlayer);
     }
 
     public void MoveToAndResize(float x, float y, int width, int height, bool movePlayer = true, bool immediate = false) {
-        if (UIController.instance.GetState() == UIController.UIState.DEFENDING) {
-            if (immediate) ArenaManager.instance.MoveToAndResizeImmediate(x, y, width, height, movePlayer);
-            else           ArenaManager.instance.MoveToAndResize(x, y, width, height, movePlayer);
-        } else
-            UnitaleUtil.DisplayLuaError("NOT THE WAVE SCRIPT", "sorry but pls don't");
+        if (immediate) ArenaManager.instance.MoveToAndResizeImmediate(x, y, width, height, movePlayer);
+        else           ArenaManager.instance.MoveToAndResize(x, y, width, height, movePlayer);
+    }
+
+    public void SetSprite(string filename) {
+        ArenaManager.instance.sprite.Set(filename);
+    }
+
+    public void SetColor(float[] color) {
+        ArenaManager.instance.sprite.color = color;
```

Comment:
I just tested this new feature, and it seems the color of the *inside* of the Arena changes, not the *outside*, which is not very good!

I think the goal here was to change the outside of the arena, while here it gives us something like this:
![like this.](https://media.discordapp.net/attachments/657012324829364276/859427390861475850/unknown.png)

Reply:
No, the feature is about changing the inner sprite

Reply:
Then we should have both, because the outer part of the Arena is the part people tend to change, not the inner part of the Arena.

Maybe with the addition of Arena.innerColor(32) and Arena.outerColor(32)!

Reply:
Done. If we are going to accept these changes, I need to open an issue, because there's a sprite related bug I can't fix

Reply:
A sprite-related bug? Care to elaborate?